### ISSUE TITLE
Twine upload error InvalidDistribution

### ISSUE THREAD

**Issue by cornhundred** [jupyter-widgets/widget-ts-cookiecutter#37]
I'm getting the following error

`
InvalidDistribution: Unknown distribution format: 'index.js'   
`

when I try and upload the bundled package using the command 

`
twine upload dist/* 
`

(following the instructions from https://github.com/jupyter-widgets/widget-ts-cookiecutter). I'm working with a minimally changed widget-ts-cookiecutter template project (that I've changed to version 0.2.0), which can be found here: https://github.com/ismms-himc/clustergrammer2

**Comment by vidartf** [comment]
Thanks for this report. This is indeed an issue with this cookie cutter. The issue arises since both the JS code and the python code outputs to the "dist" folder in the root. One solution would be to move the JS code into a subdirectory, e.g "js", with corresponding changes to other instructions.

I originally tried to flatten the structure so both JS and python could be built from the root dir (i.e. without having to `cd` into "js"). Maybe this is possible instead with a smarter JS setup.

For now, a short term solution is to: ensure npm pack does not include any python artifacts from "dist"; change the twine upload command to not use a grep, but run it again for the different bundles (sdist and wheel).

**Comment by cornhundred** [comment]
Thanks @vidartf for the quick response and feedback. I'll look into these changes, but do you think this error is actually resulting in any real problems? The package seems to be uploaded correctly to PyPI (https://pypi.org/project/clustergrammer2/)

**Comment by vidartf** [comment]
As long as you are seeing both the wheel and sdist on PyPI, you should be ok. I would also double check that the python files have not snuck into the JS distro (check tar on npm).

**Comment by cornhundred** [comment]
Thanks, I see two files on PYPI that I think correspond to the wheel and sdist (see below)

<img width="811" alt="screen shot 2018-10-23 at 11 52 32 am" src="https://user-images.githubusercontent.com/8352840/47373628-379b8400-d6ba-11e8-9a0f-b98d39fdee66.png">

I will work on checking the NPM package for python files. 

BTW, I saw the same error when I was publishing the widget Clustergrammer_Widget2 and I've been able to pip install Clustergrammer_Widget2 successfully (https://github.com/ismms-himc/clustergrammer_widget2).

I'm working on rebuilding Clustergrammer_Widget2 as Clustergrammer2 to have a shorter name and get more practice building a widget.
